pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,446 @@
+	// key: JSONName, value: verbosity
+	var verbosityVal = map[string]uint{
+		"virtAPI":        virtconfig.DefaultVirtAPILogVerbosity,
+		"virtController": virtconfig.DefaultVirtControllerLogVerbosity,
+		"virtHandler":    virtconfig.DefaultVirtHandlerLogVerbosity,
+		"virtLauncher":   virtconfig.DefaultVirtLauncherLogVerbosity,
+		"virtOperator":   virtconfig.DefaultVirtOperatorLogVerbosity,
+	}
+
+	// update the verbosity based on the existing verbosity in the KubeVirt CR
+	for key, value := range currentLv {
+		verbosityVal[key] = value
+	}
+
+	// create a message to show verbosity for the specified component
+	lines := createOutputLines(verbosityVal, options)
+
+	return lines
+}
+
+func addPatch(patchData *[]patch.PatchOperation, currentLv map[string]uint) {
+	*patchData = append(*patchData, patch.PatchOperation{
+		Op:    patchOperation,
+		Path:  patchPath,
+		Value: currentLv,
+	})
+}
+
+func setVerbosity(currentLv, options map[string]uint, patchData *[]patch.PatchOperation) {
+	// update currentLv based on the user-specified verbosity for all components
+	if verbosity, exist := options["all"]; exist {
+		for component := VirtAPI; component < All; component++ {
+			JSONName := getJSONNameByVirtComponent(component)
+			currentLv[JSONName] = verbosity
+		}
+	}
+
+	// update currentLv based on the user-specified verbosity for each component
+	for componentName, verbosity := range options {
+		if componentName == "all" {
+			continue
+		}
+		JSONName := getJSONNameByComponentName(componentName)
+		currentLv[JSONName] = verbosity
+	}
+
+	// in case of just reset (no set operation after the reset), don't need to add another patch
+	if len(currentLv) != 0 {
+		addPatch(patchData, currentLv)
+	}
+}
+
+func createPatch(currentLv, options map[string]uint) ([]byte, error) {
+	patchData := []patch.PatchOperation{}
+
+	if isReset {
+		// reset only if verbosity exists, otherwise do nothing
+		if len(currentLv) != 0 {
+			// add an empty object (removing the logVerbosity field can be another method)
+			currentLv = map[string]uint{}
+			addPatch(&patchData, currentLv)
+		}
+	}
+
+	setVerbosity(currentLv, options, &patchData)
+
+	return json.Marshal(patchData)
+}
+
+func findOperation(cmd *cobra.Command, options map[string]uint) (operation, error) {
+	isShow, isSet := false, false
+
+	for component := VirtAPI; component <= All; component++ {
+		componentName := GetComponentNameByVirtComponent(component)
+
+		// check if the flag for the component is specified
+		if !cmd.Flags().Changed(componentName) {
+			continue // do nothing for the component
+		}
+
+		// if flag is specified, it means either set or show
+		// if the value = noArg, it means show
+		// if the value != noArg, it means set
+		isShow = isShow || verbosities[component] == noArg
+		isSet = isSet || verbosities[component] != noArg
+
+		// check whether the verbosity is in the range
+		if verbosities[component] != noArg && verbosities[component] > maxVerbosity {
```

Comment:
Question: At this point, we did check if the flag was specified so we can actually error if user used `10` as value, no? Didn't try locally, but my suggestion is about changing this to just `if verbosities[component] > maxVerbosity`

Reply:
I see your point. Yes, we can return an error. However, the user can see the value 10 as the default in the help like this.
`     --all uint[=10] show/set all component log verbosity (0-9) (default 100)`

So I think it might be better to allow the user to specify `--all=10`, which equals `-all` (show operation).

If we return an error in this case, the help seems to be telling a lie.

I think it is difficult for the user to understand the meaning of the default values in the help, especially the no flag default value 100 is entirely irrelevant to the user. There is no way for the user to understand what the 100 means. So, to be honest, revealing the default values confuses the user. So it would be nice if the default values could be hidden in the help. Unfortunately, it is not possible as discussed before. So, it might be better to allow the user to specify 10 as the no argument, which means show operation. What do you think?